parse/parser.go

```diff
@@ -49,6 +49,16 @@ var datastructureMap = map[TokenType]ast.DataStructure{
 	VoidType:   ast.VoidType,
 }
 
+// identifierTypeMap maps TokenType and TokenType.
+// It is used checking identifier and assigned check
+// for example, int a = "1234" will be failed.
+var identifierTypeMap = map[TokenType]ast.DataStructure{
+	Int:    ast.IntType,
```

Comment:
I asked you how can we implement this with function's return type.(I remember...maybe..)
We should have information of all functions before parsing, but It is very difficult.
Maybe this can be implemented after parsing (compile time)..
Or lexing phase, we can do something for this work.

Reply:
In this version, function check only primitive type like int, string, bool.
so function call is not checked.